fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -617,6 +620,9 @@ public void executeByLegacy(TUniqueId queryId) throws Exception {
                     throw e;
                 } finally {
                     queryAnalysisSpan.end();
+                    if (queryQueue != null) {
```

Comment:
try (Token t = queue.offer()) 

and in token's close method, try to --concurrenty queries